### DISCUSSION TITLE
What's the correct way to upload a maintenance.tar for using with the dokku-maintenance plugin?

### DISCUSSION THREAD

**Discussion by thepuzzlemaster** [dokku/dokku#5302] [Q&A]
I've got the dokku-maintenance plugin installed and working with my dokku instance. But I'm trying to create a custom page for it, and I'm a bit unclear on how to get the .tar file up to my dokku server? What's the best-practice way to do that? I didn't see it mentioned in the docs, and wasn't quite sure what the best option was when looking over the general dokku docs either.

Thanks!

**Comment by thepuzzlemaster** [comment]
I tried using scp to push the file to my server. I put it in `/tmp/maintenance.tar`.

Then ran the following, but received the following error:

```
dokku maintenance:custom-page my-app < /tmp/maintenance.tar
/var/lib/dokku/plugins/enabled/maintenance/subcommands/custom-page: line 6: maintenance-custom-page-cmd: command not found
```

Using dokku v 0.26.0

I can enable/disable maintenance mode fine. But having some trouble with the custom page functionality.

This is the content of `/var/lib/dokku/plugins/enabled/maintenance/subcommands/custom-page`

```
#!/usr/bin/env bash
source "$PLUGIN_AVAILABLE_PATH/maintenance/command-functions"
set -eo pipefail
[[ $DOKKU_TRACE ]] && set -x

maintenance-custom-page-cmd "$@"
```

**Comment by thepuzzlemaster** [comment]
When I look through this most recent PR, I am seeing the name of this command function change from `maintenance-custom-page-cmd` to `cmd-maintenance-custom-page`. Which seems like it is likely related.
https://github.com/dokku/dokku-maintenance/pull/14/files#diff-ada0ba3b1255490d74197cffb40c6b87f4406742f1fad0f322a23c8cab953e23L7

I manually edited the file to replace `maintenance-custom-page-cmd` with `cmd-maintenance-custom-page` and the command ran, so I'm thinking this might just be an oversight/mistake from the previous PR which did the rename/refactor. Going to open a [defect](https://github.com/dokku/dokku-maintenance/issues/15) in the actual plugin repo.

However, would still love some guidance on getting the tarball up to my server. Apologies for the sidetracking.

**Reply by josegonzalez** [reply]
`scp` is probably what you want to upload a file.

**Reply by thepuzzlemaster** [reply]
Thanks for your thoughts. Is there any location that makes the most sense? I suppose that kind of depends on the specifics of the server, huh? 

I was thinking about creating a PR to update the docs with some additional guidance, but I'm a little worried that might be too specific to my setup. 

I'll create the PR, and try to keep it pretty generic and see what you think.

**Reply by josegonzalez** [reply]
I usually scp things into `/tmp`. The biggest thing is to ensure the `dokku` user has access to the file via `chown` after upload.
